inference-engine/src/vpu/graph_transformer/src/stages/split.cpp

```diff
@@ -153,18 +153,66 @@ Stage StageBuilder::addSplitStage(
         const DataVector& outputs) {
     std::vector<DimValues> offsets;
     offsets.reserve(outputs.size());
-
     DimValues curOffset({{axis, 0}});
-    for (const auto& output : outputs) {
-        offsets.emplace_back(curOffset);
-        curOffset.set(axis, curOffset[axis] + output->desc().dim(axis));
-    }
 
-    auto stage = addSplitStage(model, name, layer, std::move(offsets), input, outputs);
+    auto isAllOutputsHandled = [](const DataVector& outputs) {
+        for (int i = 0; i < outputs.size(); ++i) {
+            if (outputs[i] == nullptr) {
+                return false;
+            }
+        }
+        return true;
+    };
+
+    auto getOutAxisSizes = [&]() {
+        const auto inputDims = layer->input()->getDims();
+        const auto firstOutDims = layer->outData[0]->getDims();
+        int idx = 0;
+        for ( ; idx < inputDims.size(); ++idx) {
+            if (inputDims[idx] != firstOutDims[idx]) {
+                break;
+                }
+        }
+        std::vector<size_t> outAxisSizes;
+        outAxisSizes.reserve(inputDims.size());
+        for (const auto out : layer->outData) {
+            outAxisSizes.push_back(out->getDims()[idx]);
+        }
+        return outAxisSizes;
+    };
+
+
+    if (!isAllOutputsHandled(outputs)) {
+        const auto outAxisSizes = getOutAxisSizes();
+        auto outputs_ = outputs;
+        auto it = outAxisSizes.begin();
+        for (int i = 0; i < outputs_.size() ; ++i) {
+            if (outputs_[i] == nullptr) {
+                curOffset.set(axis, curOffset[axis] + *it++);
```

Comment:
But you do this both for this branch and below, so you may rewrite the code to do it only once.